collections/codec/bool.go

```diff
@@ -6,65 +6,100 @@ import (
 	"strconv"
 )
 
-func NewBoolKey[T ~bool]() NameableKeyCodec[T] { return boolKey[T]{} }
+// NewBoolKey returns a new boolKey codec that can handle encoding and decoding boolean values.
+// This codec can work with any type `T` that is derived from a boolean (~bool).
+func NewBoolKey[T ~bool]() NameableKeyCodec[T] {
+	return boolKey[T]{}
+}
 
+// boolKey implements the KeyCodec interface for boolean values.
+// It provides methods for encoding and decoding boolean values as keys.
 type boolKey[T ~bool] struct{}
 
+// Encode encodes the boolean value into the first byte of the buffer.
+// If the value is true, the buffer gets the byte value `0x1`, otherwise `0x0`.
+// It returns the number of bytes written (always 1) or an error if something goes wrong.
 func (b boolKey[T]) Encode(buffer []byte, key T) (int, error) {
 	if key {
-		buffer[0] = 0x1
+		buffer[0] = 0x1 // Encode true as 0x1.
 		return 1, nil
 	}
-	buffer[0] = 0x0
+	buffer[0] = 0x0 // Encode false as 0x0.
 	return 1, nil
 }
 
+// Decode reads the first byte from the buffer and converts it into a boolean value.
+// If the byte is `0x0`, it decodes to false; if `0x1`, it decodes to true.
+// It returns the number of bytes read (always 1), the decoded value, and an error if decoding fails.
 func (b boolKey[T]) Decode(buffer []byte) (int, T, error) {
 	if len(buffer) == 0 {
+		// Error if the buffer length is less than 1.
 		return 0, false, fmt.Errorf("%w: wanted size to be at least 1", ErrEncoding)
 	}
 	switch buffer[0] {
 	case 0:
+		// If the first byte is 0, return false.
```

Comment:
Let's not add very obvious comment. the code is clearly already saying that, this doesn't add anything.